src/utilities/visibility-display.scss

```
@@ -38,6 +38,52 @@
   }
 }
 
+// Show/Hide Viewport range utilities
+
+.show-whenNarrow,
+.show-whenRegular,
+.show-whenWide {
+  display: none !important;
+}
+
+.hide-whenNarrow,
+.hide-whenRegular,
+.hide-whenWide {
+  display: block !important;
+}
+
+@media (max-width: $width-md - 0.02px) {
+  .show-whenNarrow {
+    display: block !important;
+  }
+
+  .hide-whenNarrow {
+    display: none !important;
+  }
+}
+
+@media (min-width: $width-md) {
```

Something I'm wondering... should this be limited to 

```suggestion
@media (min-width: $width-md) and (max-width: $width-xl - 0.02px) {
```

so that these utilities only kick in when between the `$width-md` and `$width-xl` breakpoint and not from `$width-md` upwards.

---

@simurai That's in fact how we're considering our viewport ranges to go. Take a look at https://github.com/github/primer/issues/580#issuecomment-1076890831 to see the media queries we're proposing. You can also see a visual of these breakpoint/viewport considerations here [in this Figma frame](https://www.figma.com/file/umerHStCZ8sa2lqvateGud/Design-tokens-(FY22-Q3)?node-id=207%3A6301). (only available for hubbers) 😁

The idea is for `Narrow` to wrap all "mobile"-friendly design patterns, and `Regular` all the "desktop"-friendly ones, including wider scenarios. The `Wide` viewport range in that case is a subset of `Regular`, applied on top of it.

:v:

---

@simurai your comment made me thing of a scenario I haven't considered: when some area needs visibility on `Regular` but not on `Wide` scenarios.

I think we can support it like this: 

```html
<div class="show-whenRegular hide-whenWide">
  ...
</div>
```

I've played with CSS specificities (in a way that doesn't try to reinvent the wheel in regards to the use of `!important` in our Primer CSS utilities) [in this prototype](https://codepen.io/vdepizzol/pen/RwxKyjB?editors=1100), and have updated the PR accordingly.

Thank you! 🙇

---

I see. `Narrow` is a range (0 - 768), but `Regular` and `Wide` behave more like "breakpoints" (from ... to infinity).

Ok, yeah.. that is probably safer and you don't have to remember to add both `Regular` and `Wide`. 👍